File: .skills/openclaw-skills/skills/auxito/agentscout/src/utils/llm_client.py
"""LLM 调用封装 - OpenAI 兼容协议"""

import json
import re
from typing import Optional
from openai import OpenAI

from src.config import LLMConfig


class LLMClient:
    """通用 LLM 客户端，支持任意 OpenAI 兼容 API"""

    def __init__(self, config: LLMConfig):
        self.client = OpenAI(
            api_key=config.api_key,
            base_url=config.base_url,
        )
        self.model = config.model
        self.temperature = config.temperature
        self.max_tokens = config.max_tokens

    def chat(
        self,
        prompt: str,
        system: str = "",
        temperature: Optional[float] = None,
        max_tokens: Optional[int] = None,
    ) -> str:
        """单轮对话，返回文本"""
        messages = []
        if system:
            messages.append({"role": "system", "content": system})
        messages.append({"role": "user", "content": prompt})

        response = self.client.chat.completions.create(
            model=self.model,
            messages=messages,
            temperature=temperature or self.temperature,
            max_tokens=max_tokens or self.max_tokens,
        )
        return response.choices[0].message.content.strip()

    def chat_json(
        self,
        prompt: str,
        system: str = "",
        temperature: Optional[float] = None,
    ) -> dict:
        """对话并解析 JSON 输出"""
        raw = self.chat(prompt, system, temperature)
        # 尝试从 markdown 代码块中提取 JSON
        json_match = re.search(r"```(?:json)?\s*([\s\S]*?)```", raw)
        if json_match:
            raw = json_match.group(1).strip()
        return json.loads(raw)

File: .skills/openclaw-skills/skills/gotomanutd-dot/vibe-coding-cn/executors/llm-client.js
#!/usr/bin/env node

/**
 * LLM API 客户端
 * 
 * 统一调用接口，支持多个模型
 */

const https = require('https');

// API 配置
const API_CONFIG = {
  baseUrl: 'https://dashscope.aliyuncs.com/api/v1',
  apiKey: process.env.DASHSCOPE_API_KEY || ''
};

/**
 * 调用 LLM API
 * 
 * @param {string} prompt - 提示词
 * @param {string} model - 模型 ID
 * @param {string} thinking - 思考级别 (low/medium/high)
 * @returns {Promise<string>} AI 响应
 */
async function callLLM(prompt, model, thinking = 'medium') {
  return new Promise((resolve, reject) => {
    const postData = JSON.stringify({
      model,
      input: { messages: [{ role: 'user', content: prompt }] },
      parameters: {
        temperature: thinking === 'high' ? 0.9 : thinking === 'medium' ? 0.7 : 0.5,
        max_tokens: 8192
      }
    });
    
    const options = {
      hostname: 'dashscope.aliyuncs.com',
      port: 443,
      path: '/api/v1/services/aigc/text-generation/generation',
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${API_CONFIG.apiKey}`,
        'Content-Length': Buffer.byteLength(postData)
      }
    };
    
    const req = https.request(options, (res) => {
      let data = '';
      
      res.on('data', (chunk) => {
        data += chunk;
      });
      
      res.on('end', () => {
        try {
          const result = JSON.parse(data);
          if (result.output && result.output.text) {
            resolve(result.output.text);
          } else if (result.message) {
            reject(new Error(result.message));
          } else {
            reject(new Error('API 返回格式异常'));
          }
        } catch (error) {
          reject(new Error(`解析失败：${error.message}`));
        }
      });
    });
    
    req.on('error', (error) => {
      reject(new Error(`API 请求失败：${error.message}`));
    });
    
    req.write(postData);
    req.end();
  });
}

/**
 * 调用 LLM 并解析 JSON 响应
 */
async function callLLMJson(prompt, model, thinking = 'medium') {
  const response = await callLLM(prompt, model, thinking);
  
  try {
    // 提取 JSON
    const jsonMatch = response.match(/\{[\s\S]*\}/);
    if (!jsonMatch) {
      throw new Error('响应中未找到 JSON');
    }
    
    return JSON.parse(jsonMatch[0]);
  } catch (error) {
    throw new Error(`JSON 解析失败：${error.message}`);
  }
}

// 导出
module.exports = { callLLM, callLLMJson, API_CONFIG };
